Add support for static domain-to-IP mapping in k8s_dns_server for DNSChaos

ajoy-zero opened the pull request:
- Backend parses the input into a JSON object {"domain": "google.com", "ip": "1.2.3.4"}.
- The parsed mapping is sent to k8s_dns_server for storage and DNS fault injection.
- For a given domain, the corresponding IP is returned based on the static mapping.
- This change allows static IP resolution for specified domains during fault injection.

- [x] manual testing
<img width="1157" alt="image" src="https://github.com/user-attachments/assets/14be75ff-d5c6-4133-b1ed-446a21ff8ea7" />
<img width="583" alt="image" src="https://github.com/user-attachments/assets/87e48244-ba15-4fbe-a522-36746ec8a95a" />
<img width="578" alt="image" src="https://github.com/user-attachments/assets/301b48d4-e09b-4d11-ad53-53e6051da9a1" />

- [x] regression testing
<img width="1147" alt="image" src="https://github.com/user-attachments/assets/85b6bc98-1460-4d1c-8a13-039b54181c7c" />
<img width="556" alt="image" src="https://github.com/user-attachments/assets/f694be4f-998b-456b-bc60-a8e19e4a4dc4" />

![image](https://github.com/user-attachments/assets/7962ab97-77d7-4f5c-9c9e-f8b6bb29c899)
![image](https://github.com/user-attachments/assets/ab966a1f-786e-4a43-a2cd-236e4dbdd98a)

g1eny0ung commented on handler.go:
```diff
@@ -22,10 +21,23 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 	log.Debugf("records: %v, err: %v", records, err)
-
-	if k.needChaos(chaosPod, records, state.QName()) {
+	if k.needChaos(chaosPod, records, state.QName()) && chaosPod.Action != ActionStatic {
```
Are there any considerations not to move the logic of `chaosPod.Action != ActionStatic` into the `k.needChaos`?

Theoretically, we need to handle this part of the logic in the function to keep the code clean.

ajoy-zero replied:
The reason for not moving the chaosPod. Action != ActionStatic logic into k.needChaos() is to avoid changing the original logic, which might introduce unexpected issues or affect existing behavior.  Since this part of the code has been stable, I chose to keep the separation to ensure functional consistency and reduce the risk of regression.

Once the overall behavior is verified to be stable, we can consider refactoring this logic into k.needChaos() in a follow-up PR to improve code clarity.

g1eny0ung commented on chaos.go:
```diff
@@ -39,6 +41,12 @@ type PodInfo struct {
 }
 
+// DomainIP Domain and ip mapping
+type DomainIP struct {
```
This type is not used?

g1eny0ung commented on pb/dns.proto:
```diff
@@ -37,4 +38,10 @@ message CancelDNSChaosRequest {
   string msg = 2;
+}
+
+//定义ip 域名的映射关系
```
Please do not use Chinese as comments.

ajoy-zero replied:
Thanks for pointing it out — my apologies for including Chinese in the comments.  I've removed it and updated the code accordingly.

g1eny0ung commented on kubernetes.go:
```diff
@@ -68,6 +68,8 @@ type Kubernetes struct {
 
 	ipPodMap map[string]*PodInfo
+
+	domainAndIPMap map[string]map[string]map[string]string
```
Personally, I think it would be better to change the name to `domainIPMapByNamespacedName`. Just looking at the data structure, I can't immediately know what it includes, but changing the name would help with understanding.

ajoy-zero replied:
Thanks for the suggestion — you're absolutely right. The current name domainAndIPMap is too vague, and `domainIPMapByNamespacedName` makes the structure much clearer. I've updated the name accordingly.

ajoy-zero commented:
Really appreciate your suggestions, and apologies for the issues in the original commit.

I've made the following changes based on your comments:

Removed the unused IpDomainMap type

Replaced the Chinese comment with an English one

Renamed the variable to domainIPMapByNamespacedName for better readability
@g1eny0ung
